src/app/_hooks/useEventsCategory.ts

```diff
@@ -0,0 +1,63 @@
+import { useMemo } from 'react'
+
+import { UseCategoryProps, useCategory } from '@/hooks/useCategory'
+
+import type { EventData } from '@/types/eventTypes'
+
+import { isDateValid } from '@/utils/formatDate'
+
+import { pastEventsSetting } from '@/constants/categoryConstants'
+
+export function useEventsCategory(props: UseCategoryProps<EventData>) {
+  const { entries } = props
+
+  const results = useCategory(props)
+  const { categoryQuery, categoryCounts, categorizedResults } = results
+
+  const countsWithPastEvents = useMemo(() => {
+    const pastEventsCount = entries.filter(filterByPastEvents).length
+
+    return { ...categoryCounts, [pastEventsSetting.id]: pastEventsCount }
+  }, [categoryCounts, entries])
+
+  const resultsWithPastEvents = useMemo(() => {
+    if (categoryQuery === pastEventsSetting.id) {
+      return entries.filter(filterByPastEvents)
+    }
+
+    return categorizedResults
+  }, [categoryQuery, categorizedResults, entries])
+
+  return {
+    categoryQuery,
+    categorizedResults: resultsWithPastEvents,
+    categoryCounts: countsWithPastEvents,
+  }
+}
+
+const today = getTodayDateUTCZero()
+
+function filterByPastEvents(entry: EventData) {
+  const { startDate, endDate, slug } = entry
+
+  if (endDate) {
+    if (!isDateValid(endDate)) {
+      throw new Error(`Invalid date provided for event: ${slug}`)
+    }
+    return new Date(endDate) < today
+  }
+
+  if (!isDateValid(startDate)) {
+    throw new Error(`Invalid date provided for event: ${slug}`)
+  }
+  return new Date(startDate) < today
```

Comment:
Wondering if `endData < today` would be a better indication of a past event? If `startData < today` it could still be an ongoing event

Reply:
Totally, but `endDate` is optional on `EventData`, so not guaranteed to be present here.

For reference,  I wrote in the PR description that:

> It is programmatically calculated based on the end date [...] or the start date If no end date is available.

The function looks like this:

```ts
function filterByPastEvents(entry: EventData) {
  const { startDate, endDate, slug } = entry

  if (endDate) {
    if (!isDateValid(endDate)) {
      throw new Error(`Invalid date provided for event: ${slug}`)
    }
    return new Date(endDate) < today
  }

  if (!isDateValid(startDate)) {
    throw new Error(`Invalid date provided for event: ${slug}`)
  }
  return new Date(startDate) < today
}
```

Reply:
@barbaraperic - I've changed it for better readability to. Hopefully that clears up the confusion.

```
function filterByPastEvents(entry: EventData): boolean {
  const today = getUTCMidnightToday()
  const { startDate, endDate, slug } = entry

  if (endDate && !isDateValid(endDate)) {
    throw new Error(`Invalid endDate provided for event: ${slug}`)
  }

  if (!isDateValid(startDate)) {
    throw new Error(`Invalid startDate provided for event: ${slug}`)
  }

  const eventDate = endDate ? new Date(endDate) : new Date(startDate)
  return eventDate < today
}
```